embedding/openai.go

```diff
@@ -0,0 +1,111 @@
+package embedding
+
+import (
+	"context"
+	"strings"
+
+	"github.com/tmc/langchaingo/llms/openai"
+)
+
+// DefaultBatchSize is the default length of batches.
+const DefaultBatchSize = 512
+
+// OpenAI is the embedder using the OpenAI api.
+type OpenAI struct {
+	client *openai.LLM
+
+	StripNewLines bool
+	BatchSize     int
+}
+
+var _ Embedder = OpenAI{}
+
+// NewOpenAi creates a new OpenAI with StripNewLines set to true and batch
+// size set to 512.
+func NewOpenAI() (OpenAI, error) {
```

Comment:
not a huge fan of having the vendor-specific surface area in the top level package here but that is what the python implementation does.